app/controllers/assessments_controller.rb

```diff
@@ -0,0 +1,70 @@
+class AssessmentsController < ApplicationController
+  before_action :set_contact, only: %i[new create]
+
+  def new
+    @users = User.all
+    @roles = Role.all
+    @type = %w[log schedule].include?(type_param) ? type_param : 'log'
+
+    @need = Need.new
+    @note = Note.new
+
+    @need.status = 'complete' if @type == 'log'
+    @need.user = current_user if @type == 'log'
+  end
+
+  def create
+    authorize Need
+    @type = %w[log schedule].include?(type_param) ? type_param : 'log'
+    if @type == 'log'
+      log_assessment
+    else
+      schedule_assessment
+    end
+  end
+
+  private
+
+  def set_contact
+    @contact = Contact.find(params[:contact_id])
+  end
+
+  def log_assessment
+    @need = Need.new(assessment_params.merge(contact_id: @contact.id))
+    @note = Note.new(notes_params.merge(need: @need, category: 'phone_success'))
+    if @need.valid? && @note.valid? && @need.save && @note.save
+      redirect_to contact_path(@contact)
+      return
+    end
+
+    @users = User.all
+    @roles = Role.all
+    render :new
+  end
+
+  def schedule_assessment
+    @need = Need.new(assessment_params.merge(contact_id: @contact.id))
+    @note = Note.new
```

Comment:
I don't think you need this one, right?

Reply:
it was actually simpler to include an @note rather than dealing with lots of conditional logic in the 'errors' section, e.g. `pluralize(@need.errors.count + @note.errors.count, "error")` 

The alternative would be to wrap the errors section in a `<% if type == 'log'>` section instead if you think that's better

Reply:
Oh! No worries I just assumed it was a leftover from copying & pasting